plugins/intel-gsc/fu-igsc-plugin.c

```diff
@@ -38,9 +38,26 @@ fu_igsc_plugin_constructed(GObject *obj)
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_IGSC_OPROM_FIRMWARE);
 }
 
+static void
+fu_igsc_plugin_device_changed(FuPlugin *plugin, FuDevice *device)
+{
+	// Check for the WEDGED property in device metadata
+	const gchar *wedged = fu_device_get_metadata(device, "WEDGED");
+	if (wedged && g_strcmp0(wedged, "firmware-flash") == 0) {
```

Comment:
I think you want to use `fu_udev_device_read_property(FU_UDEV_DEVICE(device))` here -- the metadata store is really a hack for inter-plugin metadata sharing.